src/widgets/helper/ChannelView.cpp

```diff
@@ -1786,8 +1789,48 @@ void ChannelView::leaveEvent(QEvent * /*event*/)
     this->unpause(PauseReason::Mouse);
 }
 
+bool ChannelView::event(QEvent *event)
+{
+     if (event->type() == QEvent::Gesture)
+     {
+         return gestureEvent(dynamic_cast<QGestureEvent*>(event));
+     }
+
+    return BaseWidget::event(event);
+}
+
+bool ChannelView::gestureEvent(const QGestureEvent *event)
+{
+     if (QGesture *pan = event->gesture(Qt::PanGesture))
+     {
+         auto const *gesture = dynamic_cast<QPanGesture *>(pan);
+
+         switch (gesture->state())
+         {
+             case Qt::GestureStarted:
+             case Qt::GestureUpdated:
+                 this->isPanning_ = true;
+                 this->clearSelection();
+                 break;
+             default:
+                 this->isPanning_ = false;
+                 break;
+         }
+
+         this->scrollBar_->offset(-gesture->delta().y() * 0.5);
+     }
+
+     return false;
+}
+
 void ChannelView::mouseMoveEvent(QMouseEvent *event)
 {
+    if (this->isPanning_)
+    {
+        // Don't do any text selection, hovering, etc while panning
```

Comment:
How does this interact with `mousePressEvent`? Does `mousePressEvent` get called before? If so, it's worth checking that its state gets "reset", as it could set e.g. `isLeftMouseDown_`.

Reply:
I tried changing the mouse press and release event handling while panning, but it caused some selection issues when doing mouse movement after panning.